extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java

```diff
@@ -899,34 +899,40 @@ private static void setHttpServerTiming(boolean httpDisabled, HttpServerOptions
         int socketCount = 0;
 
         if (!httpDisabled && httpServerOptions != null) {
-            serverListeningMessage.append(String.format(
-                    "http://%s:%s", httpServerOptions.getHost(), actualHttpPort));
+            serverListeningMessage.append(String.format("http://%s:%s", getHost(httpServerOptions), actualHttpPort));
             socketCount++;
         }
 
         if (sslConfig != null) {
             if (socketCount > 0) {
                 serverListeningMessage.append(" and ");
             }
-            serverListeningMessage.append(String.format("https://%s:%s", sslConfig.getHost(), actualHttpsPort));
+            serverListeningMessage.append(String.format("https://%s:%s", getHost(sslConfig), actualHttpsPort));
             socketCount++;
         }
 
         if (domainSocketOptions != null) {
             if (socketCount > 0) {
                 serverListeningMessage.append(" and ");
             }
-            serverListeningMessage.append(String.format("unix:%s", domainSocketOptions.getHost()));
+            serverListeningMessage.append(String.format("unix:%s", getHost(domainSocketOptions)));
         }
         if (managementConfig != null) {
             serverListeningMessage.append(
                     String.format(". Management interface listening on http%s://%s:%s.", managementConfig.isSsl() ? "s" : "",
-                            managementConfig.getHost(), actualManagementPort));
+                            getHost(managementConfig), actualManagementPort));
         }
 
         Timing.setHttpServer(serverListeningMessage.toString(), auxiliaryApplication);
     }
 
+    /**
+     * @return The host or 'localhost' when started in dev/test mode under WSL.
+     */
+    private static String getHost(HttpServerOptions options) {
+        return (LaunchMode.current().isDevOrTest() && LaunchMode.isWSL()) ? "localhost" : options.getHost();
+    }
```

Comment:
I think this is not entirely true because if the host is defined, it won't be set to `localhost` automatically. At least that's not what `HttpHostConfigSource` is doing: it will set the host only if not defined. And also only for `quarkus.http.host` AFAICS.

@radcortez does the default value set in `HttpHostConfigSource` get returned when calling `httpServerOptions.getHost()`? I would expect it to be.

Reply:
Yes, because it is set manually here:
https://github.com/quarkusio/quarkus/blob/32e72944e0dcf4ca8e71218a42be920e502d0b33/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/options/HttpServerOptionsUtils.java#L238

Using the configuration class value.

Reply:
> I think this is not entirely true because if the host is defined, it won't be set to `localhost` automatically. At least that's not what `HttpHostConfigSource` is doing: it will set the host only if not defined. And also only for `quarkus.http.host` AFAICS.

Correct. This code is not executed if a higher ordinal configuration source sets `quarkus.http.host`.

Reply:
Here you need to also test that `options.getHost()` is `0.0.0.0` before rewriting it to `localhost`.